Source/System/GraphicalPrimitive.cpp

```diff
@@ -504,9 +504,13 @@ void BitmapPrimitive::Draw(BITMAP* drawScreen, const Vector& targetPos) {
 		return;
 	}
 
-	BITMAP* bitmapToDraw = create_bitmap_ex(8, m_Bitmap->w, m_Bitmap->h);
+	BITMAP* bitmapToDraw = create_bitmap_ex(8, (int)(m_Bitmap->w * m_Scale), (int)(m_Bitmap->h * m_Scale));
```

Comment:
```suggestion
	BITMAP* bitmapToDraw = create_bitmap_ex(8, m_Bitmap->w * m_Scale, m_Bitmap->h * m_Scale);
```
no need to explicitly cast to int. also avoid c-style casts, use static_cast<> instead